runtime/src/bank/builtins/core_bpf_migration/source_buffer.rs

```diff
@@ -0,0 +1,173 @@
+use {
+    super::error::CoreBpfMigrationError,
+    crate::bank::Bank,
+    solana_sdk::{
+        account::{AccountSharedData, ReadableAccount},
+        bpf_loader_upgradeable::{self, UpgradeableLoaderState},
+        pubkey::Pubkey,
+    },
+};
+
+/// The account details of a buffer account slated to replace a built-in
+/// program.
+#[derive(Debug)]
+pub(crate) struct SourceBuffer {
+    pub buffer_address: Pubkey,
+    pub buffer_account: AccountSharedData,
```

Comment:
I did this so that I wouldn't have to determine that offset then deserialize the elf multiple times. It happens in the checks and then again in direct deploy.

Reply:
I've updated this to just store the offset, to account for either `Some` or `None` for authority. Seems much cleaner!